setup.py

```diff
@@ -86,14 +92,19 @@ def find_javac(possible_homes):
     return name  # Fall back to "hope it's on the path"
 
 
-def compile_native_invocation_handler(*possible_homes):
+def compile_native_invocation_handler(*args):
     '''Find javac and compile NativeInvocationHandler.java.'''
-    javac = find_javac(possible_homes)
-    subprocess.check_call([
-        javac, '-target', '1.6', '-source', '1.6',
-        join('jnius', 'src', 'org', 'jnius', 'NativeInvocationHandler.java')
-    ])
-
+    javac = find_javac()
+    if JDK_HOME[-2:] == '10' or JDK_HOME[-2:] == '11':
```

Comment:
This approach isn't usable in the large scale. Will we append `or JDK_HOME[-2:] == '12':` when Java 12 comes out?